app/handlers/sessions/login/login.go

```diff
@@ -91,6 +92,7 @@ func LoginSubmitHandler(c *gin.Context) {
 		data.Password,
 		c.PostForm("password"),
 	); err != nil {
+		slog.Error("error", "Error comparing passwords: ", err.Error())
```

Comment:
some messages are formatted like `"An error has occurred: ", err.Error()` while others are `"An error has occurred!", err.Error()` - probably best to remove all punctuation - i.e. `"An error has occurred", err.Error()` as i think these will be split into separate variables within the final json object - like `{"message": "An error has occurred", "error": "int not divisible by 0"}`